src/FSharp.Analyzers.Cli/Program.fs

```diff
@@ -157,26 +159,33 @@ let runProjectAux
                         true
                     | None -> false
             )
-            |> Array.choose (fun fileName ->
+            |> Array.map (fun fileName ->
                 let fileContent = File.ReadAllText fileName
                 let sourceText = SourceText.ofString fileContent
 
                 Utils.typeCheckFile fcs logger fsharpOptions fileName (Utils.SourceOfSource.SourceText sourceText)
-                |> Option.map (Utils.createContext checkProjectResults fileName sourceText)
+                |> Result.map (Utils.createContext checkProjectResults fileName sourceText)
             )
             |> Array.map (fun ctx ->
-                logger.LogInformation("Running analyzers for {0}", ctx.FileName)
-                client.RunAnalyzers ctx
+                match ctx with
+                | Error e -> async.Return(Error e)
+                | Ok ctx ->
+                    async {
+                        logger.LogInformation("Running analyzers for {0}", ctx.FileName)
+                        let! results = client.RunAnalyzers ctx
+                        return Ok results
+                    }
             )
             |> Async.Parallel
 
         return
-            Some
```

Comment:
There was never any chance that this was `None`, so again I tightened up the types to get rid of the `option`. Instead it's now applied where it's needed (namely in the big `match` in `Program.fs`).